File: src/main/java/me/bristermitten/plumber/object/player/PPlayer.java
package me.bristermitten.plumber.object.player;

import me.bristermitten.plumber.object.builder.PlayerActionBuilder;
import me.bristermitten.plumber.object.builder.TaskLengthConfiguration;
import me.bristermitten.plumber.object.key.KeyHolder;
import org.bukkit.entity.Player;
import org.bukkit.event.Cancellable;
import org.bukkit.event.player.PlayerEvent;

public interface PPlayer extends KeyHolder {
    Player player();

    <T extends PlayerEvent & Cancellable> TaskLengthConfiguration<PlayerActionBuilder> blockEvent(Class<T> e);

    void message(String msg);
}

File: src/main/java/me/bristermitten/rewrite/dsl/player/PPlayer.kt
package me.bristermitten.rewrite.dsl.player

import me.bristermitten.rewrite.dsl.core.LengthConfiguration
import me.bristermitten.rewrite.dsl.core.PlumberEntity
import org.bukkit.event.Cancellable
import org.bukkit.event.player.PlayerEvent
import kotlin.reflect.KClass

interface PPlayer : PlumberEntity<PPlayer, PlayerActionFilter, PlayerReactorPicker> {

    fun <T> blockEvent(clazz: Class<T>): LengthConfiguration<PlayerActionFilter, PlayerReactorPicker> where T : PlayerEvent, T : Cancellable

    fun <T> blockEvent(clazz: KClass<T>): LengthConfiguration<PlayerActionFilter, PlayerReactorPicker> where T : PlayerEvent, T : Cancellable

    fun sendMessage(message: String)

    fun kill()
}
